File: python/stringy/stringy.py
#!/usr/bin/env python3
"""Read and write CSV and TSV."""
import sys


def main():
    """Main CLI function."""
    if len(sys.argv) != 3:
        print(f"Usage: {sys.argv[0]} INFILE OUTFILE")
        sys.exit(1)
    with open(sys.argv[1], "r", encoding="utf-8") as inf, \
         open(sys.argv[2], "w", encoding="utf-8") as outf:
        for line in inf:
            fields = line.rstrip("\n").split(",")
            print("\t".join(fields), file=outf)


if __name__ == "__main__":
    main()

File: cpp/stringy/stringy.cpp
#include <string>
#include <iostream>
#include <fstream>
#include <vector>

int main(int argc, char** argv) {
    if (argc != 3) {
        std::cout << "Usage: " << argv[0] << " INFILE OUTFILE" << std::endl;
        return EXIT_FAILURE;
    }
    std::ifstream inf(std::string(argv[1]), std::ios::binary);
    std::ofstream outf(std::string(argv[2]), std::ios::binary);
    std::string line;
    while (std::getline(inf, line)) {
        std::vector<std::string> fields;
        size_t lastcm = 0;
        size_t nextcm = 0;
        while ((nextcm = line.find(",", lastcm)) != std::string::npos) {
            fields.push_back(line.substr(lastcm, nextcm-lastcm));
            lastcm = nextcm + 1;
        }
        fields.push_back(line.substr(lastcm));
        bool first = true;
        for (auto& field : fields) {
            if (!first) {
                outf << "\t";
            }
            outf << field;
            first = false;
        }
        outf << "\n";
    }
    return EXIT_SUCCESS;
}
